src/app/core/shared/search/search-configuration.service.ts

```diff
@@ -194,6 +207,60 @@ export class SearchConfigurationService implements OnDestroy {
     return this.routeService.getQueryParamsWithPrefix('f.');
   }
 
+  /**
+   * Creates an observable of SearchConfig every time the configuration$ stream emits.
+   * @param configuration$
+   * @param service
+   */
+  getConfigurationSearchConfigObservable(configuration$: Observable<string>, service: SearchService): Observable<SearchConfig> {
+    return configuration$.pipe(
+      distinctUntilChanged(),
+      switchMap((configuration) => service.getSearchConfigurationFor(null, configuration)),
+      getAllSucceededRemoteDataPayload());
+  }
+
+  /**
+   * Every time searchConfig change (after a configuration change) it update the navigation with the default sort option
+   * and emit the new paginateSearchOptions value.
+   * @param configuration$
+   * @param service
+   */
+  initializeSortOptionsFromConfiguration(searchConfig$: Observable<SearchConfig>) {
+    const subscription = searchConfig$.pipe(switchMap((searchConfig) => combineLatest([
+      of(searchConfig),
+      this.paginatedSearchOptions.pipe(take(1))
+    ]))).subscribe(([searchConfig, searchOptions]) => {
+      const field = searchConfig.sortOptions[0].name;
+      const direction = searchConfig.sortOptions[0].sortOrder.toLowerCase() === SortDirection.ASC.toLowerCase() ? SortDirection.ASC : SortDirection.DESC;
+      const updateValue = Object.assign(new PaginatedSearchOptions({}), searchOptions, {
+        sort: new SortOptions(field, direction)
+      });
+      this.paginationService.updateRoute(this.paginationID,
+        {
+          sortDirection: updateValue.sort.direction,
+          sortField: updateValue.sort.field,
+        });
+      this.paginatedSearchOptions.next(updateValue);
+    });
+    this.subs.push(subscription);
+  }
+
+  /**
+   * Creates an observable of available SortOptions[] every time the searchConfig$ stream emits.
+   * @param searchConfig$
+   * @param service
+   */
+  getConfigurationSortOptionsObservable(searchConfig$: Observable<SearchConfig>): Observable<SortOptions[]> {
+    return searchConfig$.pipe(map((searchConfig) => {
+      const sortOptions = [];
+      searchConfig.sortOptions.forEach(sortOption => {
+        sortOptions.push(new SortOptions(sortOption.name, SortDirection.ASC));
+        sortOptions.push(new SortOptions(sortOption.name, SortDirection.DESC));
```

Comment:
You could add the direction here that is specified in `sortOption` instead of both